### PR TITLE
feat: optional Liger fused RMSNorm/SwiGLU kernels for DFlash training

### PR DESCRIPTION
**Pull request by jessiewei7**
## What

Adds an opt-in `model.use_liger_kernel` configuration field and a `specforge[liger]` extra. For `training.strategy=dflash`, Liger's Qwen3 RMSNorm/SwiGLU fused Triton kernels are patched in immediately before draft-model construction.

RoPE and cross-entropy patches are left disabled: DFlash owns its asymmetric rotary embedding and computes its loss outside the Transformers Qwen3 forward path. `dflash.py` resolves `Qwen3RMSNorm`/`Qwen3MLP` through the `modeling_qwen3` module at construction time, so the opt-in patch is independent of import order.

Default behavior is unchanged (`use_liger_kernel` defaults to `False`; the base install has no dependency on `liger-kernel`).

## Validation

- **CPU unit tests** (`tests/test_training/test_liger_kernel.py`, `tests/test_config/test_schema.py`): the flag defaults off, the disabled path never imports liger, the patch is applied with the frozen safe option set, non-dflash strategies are rejected before import, and a missing extra raises an actionable error.
- **CUDA numerical validation on a fixed BF16 probe**: the patched Qwen3 RMSNorm/SwiGLU forward outputs are bit-identical to the eager path, and a forward+backward pass over the patched modules yields finite outputs, a finite scalar probe loss, and finite gradients (inputs and weights).

This PR makes no new end-to-end performance claim.

### CODE REVIEW

File: scripts/train_dflash.py [hunk 1]
```diff
@@ -429,6 +435,21 @@ def main():
             print(f"Loading draft config from checkpoint: {checkpoint_config_path}")
             args.draft_config_path = checkpoint_config_path
 
+    if args.use_liger_kernel:
+        from liger_kernel.transformers import apply_liger_kernel_to_qwen3
```
[the remark this answers is not in the document]
Comment (jessiewei7) [reply scripts/train_dflash.py]: Done in 4c3e929.

**Comment by jiapingW** [comment]
@jessiewei7 Hi, I'm very sorry for the late reply. We've been refactoring the code recently, and it's mostly finished. Could you rebase your PR to the current main branch? Thanks. If your code depends on the old code, you can rebase to the `colocated` branch.

**Comment by jessiewei7** [comment]
@jiapingW Thanks for the guidance. I rebased the PR onto the current `main` branch rather than `colocated`, and migrated the integration to the config-driven `model.use_liger_kernel` path scoped to `training.strategy=dflash`.

File: specforge/modeling/draft/dflash.py
Comment (jiapingW) [inline specforge/modeling/draft/dflash.py:1]: The origin code has imported `Qwen3MLP` from `modeling_qwen3`. Why  you add module from `modeling_qwen.xxx` ?

Comment (jessiewei7) [reply specforge/modeling/draft/dflash.py:1]: Liger rebinds these classes on `modeling_qwen3`. Direct imports retain the pre-patch references, while resolving them through the module at DFlash construction time applies the opt-in patch without an import-order dependency.

Comment (jiapingW) [reply specforge/modeling/draft/dflash.py:1]: I see.

Comment (jiapingW) [reply specforge/modeling/draft/dflash.py:1]: I have refactored the code to apply liger kernel without monkey patch. Can help review it?

Comment (jessiewei7) [reply specforge/modeling/draft/dflash.py:1]: Thanks — this is a cleaner design. I reviewed the refactor and tests; LGTM.

**Review by jiapingW — APPROVED** [review]
